docs/content/en/docs/concepts/executable-playbooks.md

```diff
@@ -0,0 +1,387 @@
+
+![Example playbook flow](/SOARCA/images/example-playbook.svg)
+
+The JSON of this CACAO playbook can be found [at the bottom of this page](#example-playbook).
+
+As you can see, this playbook contains a mix of logical steps (`if-condition` and `while-condition`) and `action` steps that perform commands on a target system.
+
+### Agents and targets
+
+In CACAO playbooks, entities that execute commands are called agents, and the entities against which the commands are executed are called targets (see [Agents and Targets](https://docs.oasis-open.org/cacao/security-playbooks/v2.0/cs01/security-playbooks-v2.0-cs01.html#_Toc152256509) in the spec).
+
+Every action step in a CACAO playbook must have a single agent, and one or more targets. Both agents and targets are defined on the playbook level, in the `agent_definitions` and `target_definitions` fields. SOARCA will execute action steps that have an agent of the type `soarca`. The capability that will be selected to execute the step is determined by the `name` field of the agent. For more information, [read the documentation on components](/docs/core-components/modules).
+
+```json
+"target_definitions": {
+    "linux--b49069c2-0b69-4a46-8509-80196c4a9bf8": {
+      "type": "linux",
+      "name": "Target system",
+      "description": "System to execute commands on",
+      "address": {
+        "ipv4": [
+          "__target_ip__:value"
+        ]
+      }
+    }
+}
+```
+
+### Start and end steps
+
+Every CACAO playbook should start with a `start` step. From there, each step can define which step should be executed after the current step finishes. Depending on whether the current step has executed successfully, the next step is defined by the `on_success` or `on_failure` field. Alternatively an `on_completion` field can be defined, which identifies the next step regardless of the outcome of the current step.
+
+Example of a `start` step:
+
+```json
+"start--d6c44626-c9b6-426b-ad5d-3311bafaf068": {
+    "on_completion": "action--4b08af84-3741-48ca-8c92-df1557a87379",
+    "type": "start"
+}
+```
+
+According to the CACAO specification, every branch of steps should end in a (unique) `end` step. This is the only step type may not specify a next step:
+
+```json
+"end--60fc8d0c-3677-4363-8576-9ea9014f8c8e": {
+    "type": "end"
+}
+```
+
+### If-condition, while-condition, and parallel steps
+
+An `if-condition` step allows executing different branches depending on a specified condition. The step must specify an `on_true` field, which references the start of a branch of steps that should be executed if the condition evaluates to true. Optionally, the `if-condition` step can define an `on_false` field that defines an alternative branch that is executed if the condition evaluates to false. In each case, the specified branch keeps executing until it encounters an `end` step.
```

Comment:
@lucamrgs and @MaartendeKruijf  discussed this but this makes for difficult execution, which also makes execution paths a bit tricky we will take this up with the TC.

Reply:
Yeah I think understand your issue. It's that you need to keep track of the current branch and where it started. A `brand-end` step could then specify the head of the branch so it's simpler to move onto the next step on that level, right?